internal/provider/resource_gitlab_group_project_file_template.go

```diff
@@ -0,0 +1,133 @@
+                parent of the project defined by project_id`,
+				Type: schema.TypeInt,
+
+				// Even though there is no traditional resource to create, leave "ForceNew" as "true" so that if someone
+				// changes a configuration to a different group, the old group gets "deleted" (updated to have a value
+				// of 0).
+				ForceNew: true,
+				Required: true,
+			},
+			"file_template_project_id": {
+				Description: `The ID of the project that will be used for file templates. This project must be the direct
+				child of the project defined by the group_id`,
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+		},
+	}
+})
+
+func resourceGitLabGroupProjectFileTemplateRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	groupID := d.Get("group_id").(int)
+	group, _, err := client.Groups.GetGroup(groupID, nil, gitlab.WithContext(ctx))
+	if err != nil {
+		if is404(err) {
+			log.Printf("[DEBUG] gitlab group %d not found so removing from state", groupID)
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+	if group.MarkedForDeletionOn != nil {
+		log.Printf("[DEBUG] gitlab group %s is marked for deletion", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(fmt.Sprintf("%d", group.ID))
+	d.Set("file_template_project_id", group.FileTemplateProjectID)
+
+	return nil
+}
+
+func resourceGitLabGroupProjectFileTemplateCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	groupID := d.Get("group_id").(int)
+
+	// Creating the resource means updating the existing group to link the project to the group.
+	options := &gitlab.UpdateGroupOptions{}
+	if d.HasChanges("file_template_project_id") {
+		options.FileTemplateProjectID = gitlab.Int(d.Get("file_template_project_id").(int))
+	}
+
+	_, _, err := client.Groups.UpdateGroup(groupID, options)
+	if err != nil {
+		return diag.Errorf("could not update group after creation %d: %s", groupID, err)
+	}
+	return resourceGitLabGroupProjectFileTemplateRead(ctx, d, meta)
+}
+
+func resourceGitLabGroupProjectFileTemplateDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	groupID := d.Get("group_id").(int)
+	options := &gitlab.UpdateGroupOptions{}
+
+	_, _, err := updateGroupWithOverwrittenFileTemplateOption(client, groupID, options)
+	if err != nil {
+		return diag.Errorf("could not update group %d to remove file template ID: %s", groupID, err)
+	}
+	return resourceGitLabGroupProjectFileTemplateRead(ctx, d, meta)
+}
+
+func updateGroupWithOverwrittenFileTemplateOption(client *gitlab.Client, groupID int, options *gitlab.UpdateGroupOptions) (*gitlab.Group, *gitlab.Response, error) {
+	return client.Groups.UpdateGroup(groupID, options, func(request *retryablehttp.Request) error {
```

Comment:
Out of curiosity, what happens if `options` is `nil`? If it works I'd say way pass `nil` instead of an empty options object. WDYT? 🏓

Reply:
`nil` works exactly the same, so I've gone ahead and updated it. I don't really have an opinion in this case about which one is better, I think they're equally descriptive. 😄